src/meta/src/hummock/manager/transaction.rs

```diff
@@ -149,14 +150,32 @@ impl<'a> HummockVersionTransaction<'a> {
 
         // Append SSTs to a new version.
         for (compaction_group_id, inserted_table_infos) in commit_sstables {
+            let l0_sub_level_id = new_version_delta
+                .latest_version()
+                .levels
+                .get(&compaction_group_id)
+                .and_then(|levels| {
+                    levels
+                        .l0
+                        .sub_levels
+                        .last()
+                        .map(|level| level.sub_level_id + 1)
+                })
+                .unwrap_or(
+                    new_version_delta
+                        .latest_version()
+                        .max_sub_level_id
+                        .map(|id| id + 1)
+                        .unwrap_or(FIRST_SUB_LEVEL_ID),
```

Comment:
To be backward compatible, should we init the `max_sub_level_id` on meta startup to be the actual max sub-level id and we use the `max_sub_level_id` field only for the new sublevel's id?

Reply:
> To be backward compatible, should we init the max_sub_level_id on meta startup
This has been done in `src/storage/hummock_sdk/src/version.rs`

Reply:
Therefore, reaching line 169 indicates that the version is indeed a fresh new one.

Alternatively we can make `HummockVersion:: max_sub_level_id` u64 directly instead of Option<64>.

Reply:
I think it's ok for now, the newly introduced fields keep the Option restrictions.

When line 169 is reached, the sub_level will no longer exist in the compaciton group, so it is correct to use `FIRST_SUB_LEVEL_ID` for initialisation.

Reply:
> we use the max_sub_level_id field only for the new sublevel's id

True, it's a bug. Thanks.